### PR TITLE
feat: allow superadmins to reset cache keys for IndexedDB

### PR DESCRIPTION
**Pull request by scmmishra**
This PR allows super admins to reset cache for an account. This will force the frontend to refetch the data again on next load

CleanShot 2023-05-24 at 12 21 25@2x

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/models/concerns/cache_keys.rb [hunk 1]
```diff
@@ -4,29 +4,41 @@ module CacheKeys
...
+    keys = {}
+    self.class.cacheable_models.each do |model|
+      keys[model.name.underscore.to_sym] = fetch_value_for_key(id, model.name.underscore)
+    end
+
+    keys
   end
 
   def invalidate_cache_key_for(key)
     prefixed_cache_key = get_prefixed_cache_key(id, key)
```
Comment (sojan-official) [inline app/models/concerns/cache_keys.rb:22]:
```
 def get_prefixed_cache_key(account_id, key)
    "idb-cache-key-account-#{account_id}-#{key}"
  end
```

@scmmishra  We are expecting account_id as the argument for the cache keys right ?
but it seems like we are passing get_prefixed_cache_key(id, key)  -> id here ? which is the object id ? 

could that be the reason for original issue ?

Comment (scmmishra) [reply app/models/concerns/cache_keys.rb:22]: Let me explore this

Comment (scmmishra) [reply app/models/concerns/cache_keys.rb:22]: @sojan-official just confirmed, the ID passed here is the ID for the account

Comment (sojan-official) [reply app/models/concerns/cache_keys.rb:22]: @scmmishra  Can we add a spec to ensure that it's the correct cache that gets cleared?

Comment (scmmishra) [reply app/models/concerns/cache_keys.rb:22]: Sure, I'll do that

Comment (scmmishra) [reply app/models/concerns/cache_keys.rb:22]: @sojan-official done

**Review by sojan-official — APPROVED** [review]
- Tested locally with the console to ensure the cache is cleared. 
- Added additional controller specs to test end to end

**Pull request merged**